lib/AST/NameLookup.cpp

```diff
@@ -2434,6 +2434,49 @@ CustomAttrNominalRequest::evaluate(Evaluator &evaluator,
     }
   }
 
+  // If we have more than one attribute declaration, we have an ambiguity.
+  // So, emit an ambiguity diagnostic.
+  if (nominals.size() > 1) {
+    SmallVector<NominalTypeDecl *, 4> ambiguousCandidates;
+    // Filter out declarations that cannot be property wrappers.
+    for (auto decl : nominals) {
+      if (isa<ProtocolDecl>(decl)) {
+        continue;
+      }
+      ambiguousCandidates.push_back(decl);
+    }
+    if (ambiguousCandidates.size() > 1) {
+      auto attrName = nominals.front()->getName();
+      ctx.Diags.diagnose(attr->getLocation(),
+                         diag::ambiguous_custom_attribute_ref, attrName);
+      for (auto candidate : ambiguousCandidates) {
+        ctx.Diags.diagnose(candidate->getLoc(),
+                           diag::found_attribute_candidate);
+        // If the candidate is a top-level property wrapper, let's suggest
```

Comment:
Nothing in this code looks specific to property wrappers - will we get the nice diagnostic for ambiguous function builder attributes too?

Reply:
Yeah, you're right. I should've used "custom attribute"! And yeah, we should get a diagnostic for function builders too! I'll add a test for it as well :)

Reply:
Done ✅